src/Collector/ProxyLogTrait.php

```diff
@@ -33,7 +33,7 @@ protected function logProxy(
     /**
      * @psalm-param-out array|null $arguments
      */
-    private function processLogData(array &$arguments, mixed &$result, ?object &$error): void
+    private function processLogData(?array &$arguments, mixed &$result, ?object &$error): void
```

Comment:
```suggestion
    private function processLogData(array &$arguments, mixed &$result, ?object &$error): void
```

Variable `$arguments` is always array.

Reply:
No:

```php
if (!($this->config->getLogLevel() & ContainerInterfaceProxy::LOG_ARGUMENTS)) {
            $arguments = null;
```

Reply:
Type matters for value that passed to method only.

https://3v4l.org/c3bt3